BrightFuturesTests/ResultTests.swift

```diff
@@ -95,55 +105,55 @@ class ResultTests: XCTestCase {
         XCTAssert(r.isFailure)
         XCTAssertEqual(r.error!.domain, "DivisionByZeroError")
     }
-    
-    func testFlatMapFutureSuccess() {
```

Comment:
I don't think it is bad that the Result and Future are coupled in this way. It is definitely not a core `Result` method though, BrightFutures defines it in an `extension` because of that.

Hm, so it is not possible to create a `public extension Result` in BrightFutures? I guess a free function is the next best thing.

There are plenty of breaking changes already (every change in the tests is), so I am not against it because of that.